lib/p2p/packets/Packet.ts

```diff
@@ -81,12 +82,31 @@ abstract class Packet<T = any> implements PacketInterface {
 
       if (bodyOrPacket) {
         this.body = bodyOrPacket;
-        this.header.hash = MD5(JSON.stringify(bodyOrPacket)).toString(CryptoJS.enc.Base64);
+        this.header.hash = this.hash(bodyOrPacket);
       }
     }
   }
   public abstract serialize(): Uint8Array;
 
+  private hash(value: any): string {
+    return MD5(stringify(value)).toString(CryptoJS.enc.Base64);
```

Comment:
It might be nice to have some sanity tests for this method.

Reply:
You mean to test against static values (input/output)?

Reply:
I'll add some testing in the upcoming #756. Merging this now due to dependency in #756.